client/client.go

```diff
@@ -173,8 +178,6 @@ type Config struct {
 
 func loadServerTrust(conf Config) (*x509.CertPool, error) {
 	if conf.ServerCert == "" {
-		// TODO: this is for pre-production version only to simplify tests.
-		// Make sure to remove in production version.
 		log.Warn("Server certificate not provided. Trusting all servers.")
 		return nil, nil
```

Comment:
What is our intention with `loadServerTrust`? My understanding is that the majority of hosted Mender users will likely want to run without a custom provided certificate, so it should revert to the system certificate pool in this case. Not revert to trusting everything, as seems to be done here. This might be a security problem.

What's your view on this?

Reply:
This is definitely something we should fix, and you are right. We shouldn't trust everything in this case. I will make a separate ticket for fixing this one and will leave the `TODO` as it was.